src/constructors.jl

```diff
@@ -31,45 +90,56 @@ Constructs a linear operator from a symmetric matrix. If
 its elements are real, it is also Hermitian, otherwise complex
 symmetric.
 """
-LinearOperator(M::Symmetric{T}) where {T} =
-  LinearOperator(M; symmetric = true, hermitian = eltype(M) <: Real)
+function LinearOperator(M::Symmetric{T}; storagetype = Vector{T}, kwargs...) where {T}
+  nrow, ncol = size(M)
+  Mv = storagetype(undef, nrow)
+  hermitian = eltype(M) <: Real
+  Maw = hermitian ? Mv : storagetype(undef, ncol)
+  LinearOperator(Mv, Mv, Maw, M, symmetric = true, hermitian = hermitian)
+end
 
 """
     LinearOperator(M)
-
+    
 Constructs a linear operator from a Hermitian matrix. If
 its elements are real, it is also symmetric.
 """
-LinearOperator(M::Hermitian{T}) where {T} =
-  LinearOperator(M; symmetric = eltype(M) <: Real, hermitian = true)
+function LinearOperator(M::Hermitian{T}; storagetype = Vector{T}, kwargs...) where {T}
+  nrow, ncol = size(M)
+  Mv = storagetype(undef, nrow)
+  symmetric = eltype(M) <: Real
+  Mtu = symmetric ? Mv : storagetype(undef, ncol)
+  LinearOperator(Mv, Mtu, Mv, M, symmetric = symmetric, hermitian = true)
+end
 
 # the only advantage of this constructor is optional args
 # use LinearOperator{Float64} if you mean real instead of complex
 """
     LinearOperator(nrow, ncol, symmetric, hermitian, prod,
                     [tprod=nothing,
                     ctprod=nothing])
-
 Construct a linear operator from functions.
 """
 function LinearOperator(
```

Comment:
We should probably remove this constructor and leave only the version with the type.